cmd/podman/ps.go

```diff
@@ -418,35 +421,49 @@ func getTemplateOutput(containers []*libpod.Container, opts psOptions) ([]psTemp
 			if err != nil {
 				return errors.Wrapf(err, "unable to obtain container exit code")
 			}
-			pid, err = c.PID()
+			startedTime, err = c.StartedTime()
 			if err != nil {
-				return errors.Wrapf(err, "unable to obtain container pid")
+				logrus.Errorf("error getting started time for %q: %v", c.ID(), err)
 			}
+
+			if !opts.size && !opts.namespace {
+				return nil
+			}
+
+			if opts.namespace {
+				pid, err = c.PID()
+				if err != nil {
+					return errors.Wrapf(err, "unable to obtain container pid")
+				}
+				ns = getNamespaces(pid)
```

Comment:
I think you can move this after the `Batch()` inside another `if opts.namespace` - it doesn't call into the container at all, I think.

Reply:
fixed